working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1885,14 +1900,27 @@ Where `<keyword>` is `var` or `final` is treated like so:
     }
     ```
 
-#### Pattern-if statement
+#### If-case statement
 
 1.  Evaluate the `expression` producing `v`.
 
-2.  Match the `pattern` against `v`.
+2.  Match the `pattern` in the `caseHead` against `v`.
+
+3.  If the match succeeds:
+
+    1.  If there is a guard clause:
+
+        1.  Evaluate it. If it does not evaluate to a Boolean, throw a runtime
+            exception. *This can happen if the guard expression's type is
```

Comment:
exception -> error.

In practice, this should be a completely normal failing implicit downcast from `dynamic` to `bool`.

Reply:
Done.